src/todos.rs

```diff
@@ -0,0 +1,58 @@
+use crate::terminal::{Terminal, TerminalError};
+use crate::todo::Todo;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct Todos {
+    all_todos: HashMap<u8, Todo>,
+    length: u8,
+}
+
+impl Todos {
+    pub fn new() -> Self {
+        Todos {
+            all_todos: HashMap::<u8, Todo>::new(),
+            length: 0,
+        }
+    }
+
+    pub fn insert_todo(&mut self, todo: Todo) {
+        self.length += 1;
+        self.all_todos.entry(self.length).or_insert(todo);
+    }
+
+    pub fn update(&mut self, id: u8, new_todo: Todo) -> bool {
+        if let Some(_) = self.all_todos.get(&id) {
+            self.all_todos.insert(id, new_todo);
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn show_all_todos(&mut self, show_keys: bool) -> Result<(), TerminalError> {
+        let mut terminal = Terminal::new();
+        let todos = &self.all_todos;
+        let mut keys: Vec<&u8> = todos.keys().collect();
+        keys.sort();
+
+        for key in keys {
+            if let Some(message) = todos.get(key) {
```

Comment:
Teoricamente, esse `if let Some(_)` aqui não é necessário, concorda? Isso porque você está iterando sobre as _chaves_ que o próprio mapa retornou, logo acima; então você _sabe_ que todas as chaves estarão ali.

Para eliminar esse `if`, uma opção é iterar sobre cada _entry_ do mapa. Uma _entry_ é composta pela chave _e_ pelo valor. Todavia, em `HashMap`s, esses _entries_ seriam retornados em uma ordem qualquer, já que o HashMap não garante nenhuma ordem em relação à iteração.

A `BTreeMap`, entretanto, garante que a iteração (seja pelas chaves, pelas _entries_, etc) sempre irá ocorrer de acordo com a ordem natural das chaves, o que é exatamente o que você precisa aqui. :)

Você pode usar uma referência ao mapa direto no `for`, já que ele é um iterável (i.e., implementa a trait `IntoIterator`)! [Ref.](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#impl-IntoIterator-for-%26%27a%20BTreeMap%3CK%2C%20V%2C%20A%3E)

Reply:
Traz uma vantagem grande mesmo o uso do `BTreeMap` e ali não era necessario o `if let Some(_)`, tinha me passado despercebido.